http.c

```diff
@@ -565,7 +565,7 @@ evhttp_maybe_add_date_header(struct evkeyvalq *headers)
 {
 	if (evhttp_find_header(headers, "Date") == NULL) {
 		char date[50];
-		if (sizeof(date) - evutil_date_rfc1123(date, sizeof(date), NULL) > 0) {
+		if (sizeof(date) > evutil_date_rfc1123(date, sizeof(date), NULL)) {
```

Comment:
You are right, in case of evutil_date_rfc1123 returns -1 the check works as `!=0`, but when it returns something >= 50 the the check is correct.
However this fix does not changes a lot, since `50>-1 == false`, so explicit cast is required to address this correctly.

Reply:
Also can you please update the commit subject to reflect the change (right now it is simply `Update http.c` which does not meaningful enough)

Reply:
thanks for the answer.
I was probably too stingy with the description.
maybe an example will give more information.
please see it.
```cpp
unsigned long A;

// A= 70; (A - 11) = 59; true
// A= 11; (A - 11) = 0; false
// A= 1; (A - 11) = 18446744073709551606; true
// A= 0; (A - 11) = 18446744073709551605; true
if (A - 11 > 0)
```
in my opinion, he says that any comparison is good `!=`

Reply:
> Also can you please update the commit subject to reflect the change (right now it is simply `Update http.c` which does not meaningful enough)

I'll do it later.
if I name it PR will it be okay?

Reply:
>if I name it PR will it be okay?

Yes

Reply:
>in my opinion, he says that any comparison is good !=

Ok, I see, you are right, once the right argument is greater then left, it works incorrectly

Reply:
But it is still not strictly correct in case when evutil_date_rfc1123() returns -1, can you add explicit cast for this case?

Reply:
I suggest trying it like this.